### PR TITLE
fix(MesosStateStore): add linearBackoff retry to the stream

### PR DESCRIPTION
**Pull request by orlandohohmeier**
Adjust the `MesosStateStore` to use a linear backoff strategy when re-connecting to Mesos (`retry`), as this otherwise leads to a rather high CPU load and an unresponsive UI due to a rather high number of connection attempts.

Closes DCOS-22301

### CODE REVIEW

**Comment by orlandohohmeier** [comment]
😢 build failed due to a dependency miss-match...

**Comment by juliangieseke** [comment]
fixed shrinkwrap 👍  – 🤞 again

File: package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
   },
   "dependencies": {
     "@dcos/http-service": "0.2.1",
-    "@dcos/mesos-client": "0.1.10",
+    "@dcos/mesos-client": "0.2.0",
```
Comment (nLight) [inline package.json]: We might want bumping that one to `0.2.1` to include stuff we have in `http-client@0.3.0` which is pretty important https://github.com/dcos-labs/http-service/compare/v0.2.1...v0.3.0

Comment (orlandohohmeier) [reply package.json]: @nLight could you please open a PR to update the dependency? https://jira.mesosphere.com/browse/DCOS_OSS-2400

Comment (nLight) [reply package.json]: https://github.com/dcos/dcos-ui/pull/2922

**Pull request merged**